### PR TITLE
DATAUP-747: get ready for multi select

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

Some minor changes to ready the world for select elements with multi select capabilities.

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/DATAUP-747
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and seeing that nothing has changed. How boring!

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: kbase-extension/static/kbase/js/common/sdk.js [hunk 1]
```diff
@@ -38,6 +38,15 @@ define(['common/props'], (Props) => {
         if (converted.multipleItems) {
             return [];
         }
+        try {
+            const isMultiSelect = converted.original.dropdown_options.multiselection;
+            if (isMultiSelect === 1) {
+                return [];
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/sdk.js:44]: null value: return an empty list if the element is a multi select dropdown

File: kbase-extension/static/kbase/js/common/sdk.js [hunk 2]
```diff
@@ -53,10 +62,6 @@ define(['common/props'], (Props) => {
         })();
     }
 
-    function updateNullValue(converted) {
-        converted.data.nullValue = nullValue(converted);
-    }
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/sdk.js:58]: I was getting annoyed with these functions that just redirect to another function, so replaced them with a direct call.

File: kbase-extension/static/kbase/js/common/sdk.js [hunk 3]
```diff
@@ -121,10 +134,6 @@ define(['common/props'], (Props) => {
         return defaultToNative(converted, defaultValues[0]);
     }
 
-    function updateDefaultValue(converted, spec) {
-        converted.data.defaultValue = defaultValue(converted, spec);
-    }
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/sdk.js:126]: another pointless function

File: test/unit/spec/appWidgets/validationSpec.js [hunk 4]
```diff
@@ -425,9 +425,9 @@ define([
                                     options.required = required;
                                     return Validation[method](test.value, options);
                                 }).then((result) => {
-                                    Object.keys(test.result).forEach((key) => {
-                                        expect(result[key]).toEqual(test.result[key]);
-                                    });
+                                    // ensure the result contains everything
+                                    // in test.result
+                                    expect(result).toEqual(jasmine.objectContaining(test.result));
```
Comment (ialarmedalien) [inline test/unit/spec/appWidgets/validationSpec.js:430]: use native jasmine matcher

File: test/unit/spec/appWidgets/validationSpec.js [hunk 5]
```diff
@@ -804,7 +804,7 @@ define([
                     diagnosis: Constants.DIAGNOSIS.INVALID,
                     errorMessage: 'value must be a string or number (it is of type "undefined")',
                     value: undefined,
-                    pasedValue: undefined,
+                    parsedValue: undefined,
```
Comment (ialarmedalien) [inline test/unit/spec/appWidgets/validationSpec.js:807]: found these typos when switching to the native jasmine matcher!

**Review by briehl — APPROVED** [review]

**Pull request merged**